tls/extensions/s2n_server_alpn.c

```diff
@@ -22,44 +22,48 @@
 
 #include "tls/extensions/s2n_server_alpn.h"
 
-
-/* Precalculate size of extension */
-int s2n_server_extensions_alpn_send_size(struct s2n_connection *conn)
+static bool s2n_alpn_should_send(struct s2n_connection *conn);
+static int s2n_alpn_send(struct s2n_connection *conn, struct s2n_stuffer *out);
+static int s2n_alpn_recv(struct s2n_connection *conn, struct s2n_stuffer *extension);
+
+const s2n_extension_type s2n_server_alpn_extension = {
+    .iana_value = TLS_EXTENSION_ALPN,
+    .is_response = true,
+    .send = s2n_alpn_send,
+    .recv = s2n_alpn_recv,
+    .should_send = s2n_alpn_should_send,
+    .if_missing = s2n_extension_noop_if_missing,
+};
+
+static bool s2n_alpn_should_send(struct s2n_connection *conn)
 {
-    const uint8_t application_protocol_len = strlen(conn->application_protocol);
-
-    if (!application_protocol_len) {
-        return 0;
-    }
-
-    return 3 * sizeof(uint16_t) + 1 * sizeof(uint8_t) + application_protocol_len;
+    return conn && strlen(conn->application_protocol) > 0;
```

Comment:
Should we add a check for maximum length?
https://github.com/awslabs/s2n/blob/master/tls/s2n_protocol_preferences.c#L36

Reply:
Not to should_send, but i could add it to recv.

Reply:
Actually, it's unnecessary. The protocol name length value is a uint8_t, which can't be larger than 255.